043-tasks_queue/proposal.md

```diff
@@ -0,0 +1,145 @@
+# Summary
+
+We propose to enhance Concourse's task scheduling process. Instead of tasks immediately picking a compatible worker, a task is passed to a new component: the Resource Pool. 
```

Comment:
I feel like "Resource Pool" is slightly misleading. It seems like what is happening is that ready-to-run tasks are being buffered until resources are available. "Resource Pool" to me sounds more worker-centric than task-centric.

Reply:
Yes, in this instance the "resource" is the worker (alas "workers pool" is already in use in the code base).

Reply:
```
WorkerAccess.Request()
WorkerAccess.Release()
```
🤷‍♂️ 

Do agree that resource is already overloaded